sshd-core/src/main/java/org/apache/sshd/common/util/logging/LoggingUtils.java

```diff
@@ -555,108 +545,4 @@ public String toString() {
-     */
-    public static void logExceptionStackTrace(Logger logger, Level level, Throwable t) {
-        logExceptionStackTrace(logger, level, t, getDefaultStackTraceLoggingDepth());
-    }
-
-    /**
-     * Logs the stack trace of the exception up to specified depth or available stack trace elements.
-     *
-     * @param logger The {@link Logger} instance to log the information
-     * @param level The logging {@link Level} to use
-     * @param t The {@link Throwable} data to write - ignored if {@code null}
-     * @param maxDepth Maximum stack trace elements to log - if non-positive then nothing is logged
-     */
-    public static void logExceptionStackTrace(Logger logger, Level level, Throwable t, int maxDepth) {
-        if ((t == null) || (maxDepth <= 0) || (!isLoggable(logger, level))) {
-            return;
-        }
-
-        Consumer<String> executor = loggingClosure(logger, level);
-        logExceptionStackTrace(t, maxDepth, executor);
-    }
-
-    /**
-     * Logs the stack trace of the exception up to specified depth or available stack trace elements.
-     *
-     * @param t The {@link Throwable} data to write - ignored if {@code null}
-     * @param maxDepth Maximum stack trace elements to log - if non-positive then nothing is logged
-     * @param executor The {@link Consumer} invoked for each formatted stack trace element
-     */
-    public static void logExceptionStackTrace(Throwable t, int maxDepth, Consumer<? super String> executor) {
-        if ((t == null) || (maxDepth <= 0) || (executor == null)) {
-            return;
-        }
-        StackTraceElement[] elements = t.getStackTrace();
-        int numElements = GenericUtils.length(elements);
-        StringBuilder workBuf = new StringBuilder(Byte.MAX_VALUE);
-        for (int index = 0, maxElements = Math.min(maxDepth, numElements); index < maxElements; index++) {
-            StackTraceElement ste = elements[index];
-            workBuf.setLength(0); // re-use
-            try {
-                appendStackTraceElement(workBuf.append("    at "), ste);
-            } catch (IOException e) {
-                throw new RuntimeException("Unexpected failure (" + e.getClass().getSimpleName() + ")" + " to append stack-trace-element=" + ste + ": " + e.getMessage(), e);
-            }
-            executor.accept(workBuf.toString());
-        }
-    }
-
-    /**
-     * @return The default value used by {@link #logExceptionStackTrace(Logger, Level, Throwable)}
-     */
-    public static int getDefaultStackTraceLoggingDepth() {
-        return Math.max(DEFAULT_STACK_TRACE_LOGGING_DEPTH_VALUE_HOLDER.get(), DEFAULT_STACK_TRACE_LOGGING_DEPTH_VALUE);
-    }
-
-    /**
-     * @param value The value to set - <B>Note:</B> the effective value is the <U>maximum</U>
-     * between it and {@value #DEFAULT_STACK_TRACE_LOGGING_DEPTH_VALUE}
-     */
-    public static void setDefaultStackTraceLoggingDepth(int value) {
```

Comment:
Perhaps all these utility methods can't actually be removed. I'm not sure how "API" they are. If they're public API, I'm happy to restore them all (and maybe mark them deprecated?) if you prefer. Just let me know.